features/step-definitions/sqlup-mode.el-steps.el

```diff
@@ -51,3 +51,8 @@
         (cl-letf (((symbol-function 'set-process-sentinel) #'ignore)
                   ((symbol-function 'get-buffer-process) #'ignore))
           (sql-interactive-mode))))
+
+(When "I type a newline"
+      "Simulate typing a newline character"
+      (lambda ()
+        (execute-kbd-macro (string-to-vector "\n"))))
```

Comment:
No, using keyboard macros to simulate actually pressing enter is slightly better for end-to-end testing because it is closer to what the user will actually do. In particular if a major mode binds RET to something else which breaks the code this test will fail, but if we just ran `(newline)` it would pass anyway.

Also I pretty much just copied this from [espud's `When I type ...`](https://github.com/ecukes/espuds/blob/master/espuds.el#L314) :stuck_out_tongue: (for some reason doing `When I type "select\n"` literally inserts the `\n`, quoting in ecukes is a bit flaky).